### PR TITLE
WIP: Warehouse tax

### LINKED ISSUE #112: Implement the tax for deposit ALZ in warehouse and fix the sync between client and db

### PR DESCRIPTION
**Pull request by yveltal-fan**
Current issue, sales tax min/max scales based on level, not sure what the formula is.

Fixes #112

### CODE REVIEW

File: WorldSvr/OnWarehouse.c [hunk 1]
```diff
@@ -39,24 +39,56 @@ CLIENT_PROCEDURE_BINDING(WAREHOUSE_CURRENCY_DEPOSIT) {
...
 
-	// TODO: Calculate the tax for the deposit
-
+	// Deposit
 	if (Packet->Amount > 0) {
-		if (Character->Data.Info.Alz< Packet->Amount) {
+		// TODO: Move tax to config?
+		Int64 MinimumTax = 1000;
+		Int64 MaximumTax = 10000;
+		double TaxPercent = 0.4;
```
Comment (notestsoft) [inline WorldSvr/OnWarehouse.c:47]: Use the type `Float64` instead

File: WorldSvr/OnWarehouse.c [hunk 2]
```diff
@@ -39,24 +39,56 @@ CLIENT_PROCEDURE_BINDING(WAREHOUSE_CURRENCY_DEPOSIT) {
...
+
+		Int64 TaxAmount = ((TaxPercent / 100) * Packet->Amount);
+
+		// TODO: Replace with min/max macro
+		if (TaxAmount < MinimumTax) {
```
Comment (notestsoft) [inline WorldSvr/OnWarehouse.c:52]: We have MIN(A, B) MAX(A, B) as macro available

**Review by notestsoft — APPROVED** [review]

File: WorldSvr/OnWarehouse.c [hunk 3]
```diff
@@ -65,10 +97,6 @@ CLIENT_PROCEDURE_BINDING(WAREHOUSE_CURRENCY_DEPOSIT) {
 
 	SocketSend(Socket, Connection, Response);
 	return;
-
-error:
```
Comment (notestsoft) [inline WorldSvr/OnWarehouse.c:69]: You can replace this error: label with your SocketSend(Socket, Connection, Response); with Result = 0 so you can keep using goto error.

File: WorldSvr/OnWarehouse.c [hunk 4]
```diff
@@ -65,10 +97,6 @@ CLIENT_PROCEDURE_BINDING(WAREHOUSE_CURRENCY_DEPOSIT) {
...
-	memcpy(&Character->Data.WarehouseInfo, &kWarehouseInfoBackup, sizeof(struct _RTCharacterWarehouseInfo));
```
Comment (notestsoft) [inline WorldSvr/OnWarehouse.c:70]: Delete the one on the func entry also its useless
`memcpy(&kWarehouseInfoBackup, &Character->Data.WarehouseInfo, sizeof(struct _RTCharacterWarehouseInfo));`

**Review by notestsoft — APPROVED** [review]

**Comment by yveltal-fan** [comment]
Accidentally added last commit xd

**Pull request merged**